### ISSUE TITLE
[2.x] Using onEmpty causes "runtime error: invalid memory address or nil pointer dereference" in CLI

### ISSUE THREAD

**Issue by ziebelje** [influxdata/influxdb#17263]
__Steps to reproduce:__
Example source data. This is brought in by a Scraper.
```
# HELP users_active_hour_total Total number of active users in the past hour.
# TYPE users_active_hour_total gauge
users_active_hour_total 48 1584100800000
# HELP users_active_day_total Total number of active users in the past day.
# TYPE users_active_day_total gauge
users_active_day_total 482 1584057600000
# HELP users_active_month_total Total number of active users in the past month.
# TYPE users_active_month_total gauge
users_active_month_total 3201 1583020800000
# HELP api_calls_total Number of API calls made in the past hour.
# TYPE api_calls_total gauge
api_calls_total{source="app",error="false"} 2403 1584100800000
api_calls_total{source="app",error="true"} 0 1584100800000
api_calls_total{source="cron",error="false"} 1198 1584100800000
api_calls_total{source="cron",error="true"} 10 1584100800000
api_calls_total{source="external",error="false"} 12 1584100800000
api_calls_total{source="external",error="true"} 0 1584100800000
```

Run `influx repl` on the CLI

Run this query (it works)
```
> from(bucket: "app")
  |> range(start:-6h)
  |> filter(fn: (r) => r._measurement == "api_calls_total")
  |> filter(fn: (r) => r.source == "app")
  |> filter(fn: (r) => r.error == "false")
Result: _result
Table: keys: [_start, _stop, _field, _measurement, error, source]
                   _start:time                      _stop:time           _field:string     _measurement:string            error:string           source:string                      _time:time                  _value:float
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ----------------------------
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T12:00:00.000000000Z                          2403
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T13:00:00.000000000Z                          3411
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T14:00:00.000000000Z                          5065
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T15:00:00.000000000Z                          5182
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T16:00:00.000000000Z                          5399
2020-03-13T11:50:41.295902592Z  2020-03-13T17:50:41.295902592Z                   gauge         api_calls_total                   false                     app  2020-03-13T17:00:00.000000000Z                          5871
```

Run this query (it fails):
```
> from(bucket: "app")
  |> range(start:-6h)
  |> filter(fn: (r) => r._measurement == "api_calls_total")
  |> filter(fn: (r) => r.source == "app")
  |> filter(fn: (r) => r.error == "false", onEmpty: "keep")
Error: panic: runtime error: invalid memory address or nil pointer dereference
```

However, if I paste that same query into the GUI, it works. Here's the HTTP request and response for that:
```
{"query":"from(bucket: \"app\")\r\n  |> range(start:-6h)\r\n  |> filter(fn: (r) => r._measurement == \"api_calls_total\")\r\n  |> filter(fn: (r) => r.source == \"app\")\r\n  |> filter(fn: (r) => r.error == \"false\", onEmpty: \"keep\")","extern":{"type":"File","package":null,"imports":null,"body":[{"type":"OptionStatement","assignment":{"type":"VariableAssignment","id":{"type":"Identifier","name":"v"},"init":{"type":"ObjectExpression","properties":[{"type":"Property","key":{"type":"Identifier","name":"timeRangeStart"},"value":{"type":"UnaryExpression","operator":"-","argument":{"type":"DurationLiteral","values":[{"magnitude":6,"unit":"h"}]}}},{"type":"Property","key":{"type":"Identifier","name":"timeRangeStop"},"value":{"type":"CallExpression","callee":{"type":"Identifier","name":"now"}}}]}}}]},"dialect":{"annotations":["group","datatype","default"]}}
```

```
#group,false,false,true,true,false,false,true,true,true,true
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string,string,string,string
#default,_result,0,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,,,gauge,api_calls_total,true,app
,result,table,_start,_stop,_time,_value,_field,_measurement,error,source

#group,false,false,true,true,false,false,true,true,true,true
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string,string,string,string
#default,_result,,,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement,error,source
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T12:00:00Z,2403,gauge,api_calls_total,false,app
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T13:00:00Z,3411,gauge,api_calls_total,false,app
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T14:00:00Z,5065,gauge,api_calls_total,false,app
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T15:00:00Z,5182,gauge,api_calls_total,false,app
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T16:00:00Z,5399,gauge,api_calls_total,false,app
,,1,2020-03-13T11:53:11.989895655Z,2020-03-13T17:53:11.989895655Z,2020-03-13T17:00:00Z,5871,gauge,api_calls_total,false,app
```

More error detail from my logs:
```
Mar 13 13:34:22 log influxd[24951]: ts=2020-03-13T13:34:22.027312Z lvl=info msg="Dispatcher panic" log_id=0LWpOEDW000 service=storage-reads component=dispatcher error="panic: runtime error: invalid memory address or nil pointer dereference" stacktrace="goroutine 75512 [running]:
runtime/debug.Stack(0xc0046efa40, 0x2c33700, 0x1d09a23)
  /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1.1(0xc0046efaa0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/dispatcher.go:81 +0x192
panic(0x1a00e80, 0x44cc270)
  /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/influxdata/flux/compiler.(*compilerScope).LocalRange(0xc004734b20, 0xc004734d70)
  <autogenerated>:1 +0x2e
github.com/influxdata/flux/values.BuildExternAssignments(0x2c7f7e0, 0xc0046ef1a0, 0x7f1a7ee57130, 0xc004734b20, 0xc004734b20, 0x41285e)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/values/scope.go:193 +0x50
github.com/influxdata/flux/compiler.Compile(0x2ca0ce0, 0xc004734b20, 0xc0046ef1a0, 0x2c90060, 0xc00d6ecd38, 0x19c5360, 0xc004ced980, 0xc00473a5a0, 0xc00473a5a0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/compiler/compiler.go:18 +0x297
github.com/influxdata/flux/compiler.(*CompilationCache).Compile(0xc0046f9760, 0x2c90060, 0xc00d6ecd38, 0x1, 0xc00473a628, 0x2c771a0, 0xc004ced410)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/compiler/compiler.go:422 +0xc5
github.com/influxdata/flux/execute.(*dynamicFn).prepare(0xc0046efb60, 0xc0046f6a80, 0x8, 0x8, 0xc004ced980, 0xc00ad344e0, 0xc00b3af600)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/row_fn.go:57 +0x2d3
github.com/influxdata/flux/execute.(*RowPredicateFn).Prepare(0xc0046efb60, 0xc0046f6a80, 0x8, 0x8,0x0, 0x0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/row_fn.go:191 +0x58
github.com/influxdata/flux/stdlib/universe.(*filterTransformation).Process(0xc004ced200, 0x2a5004a6b8f3226f, 0xf9470924544f98b5, 0x7f1a7a9d6760, 0xc0046f6cc0, 0x1, 0x0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/stdlib/universe/filter.go:183 +0x61
github.com/influxdata/flux/execute.processMessage(0x2c771a0, 0xc004cecde0, 0x2c80820, 0xc004ced200, 0x2c53920, 0xc0046f9d20, 0xc000199e9a, 0x17b3, 0x0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/transport.go:206 +0x2d6
github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages(0xc0046efbc0, 0x2c771a0, 0xc004cecde0, 0xa)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/transport.go:159 +0xc6
github.com/influxdata/flux/execute.(*poolDispatcher).run(0xc0046efaa0, 0x2c771a0, 0xc004cecde0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/dispatcher.go:132 +0x65
github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1(0xc0046efaa0, 0x2c771a0, 0xc004cecde0)
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/dispatcher.go:86 +0xcd
created by github.com/influxdata/flux/execute.(*poolDispatcher).Start
  /go/pkg/mod/github.com/influxdata/flux@v0.61.0/execute/dispatcher.go:63 +0x7e
"
```

__Environment info:__

* System info: Linux 4.15.0-66-generic x86_64
* InfluxDB version: InfluxDB 2.0.0-beta.5 (git: 2295247a5) build_date: 2020-03-13T11:22:29Z

__Config:__
It's a fresh installation; haven't changed anything.

**Comment by jademcgough** [comment]
There's an effort to switch over to using [Algorithm W](https://en.wikipedia.org/wiki/Hindley%E2%80%93Milner_type_system); once that work is complete, this should be resolved.

**Comment by russorat** [comment]
Thanks for the issue. I'm going to close this for now since it was opened against a pre-release version of InfluxDB. If you are still facing this problem after upgrading to the latest release, please open a new issue with the details.
